### ISSUE TITLE
SoS not sending updated workflow

### ISSUE THREAD

**Issue by pgcudahy** [vatlab/sos#1435]
Thanks very much for your help with #1434, but I've noticed a somewhat similar issue. When I'm submitting workflows to a cluster and I subtly update a workflow `task` statement, it is still submitted with the older values. For example I had a step where a few substeps were timing out on the cluster so I changed the step's task statement from `task: walltime='00:30:00', mem='6G', trunk_size=40, trunk_workers=10` to `task: walltime='00:45:00', mem='6G', trunk_size=40, trunk_workers=10` but the update workflow still had the same task id as before and on the cluster still had a walltime of 30 minutes. If I ran `sos purge` on the task and resubmitted, then the walltime would be updated. The full cell would look like this
```
%run call_variants -q yale_hpc_task_spooler -r yale_hpc_task_spooler mem="4GB" cores=1 walltime="12:00:00" nodes=1 -e ignore

[call_variants]
input: output_from("print_reads"), group_by=1
output: f'/home/pgc29/scratch60/helen_mixed_infection/data/vcf/\
{os.path.split(_input[0].resolve())[1].split(".bam")[0]}.vcf.gz'
task: walltime='00:30:00', mem='3G', trunk_size=40, trunk_workers=10, 
    workdir='/home/pgc29/scratch60/helen_mixed_infection/data/vcf'
run: expand=True
    module load BCFtools
    bcftools mpileup {_input} --threads 1 -AI \
    -f /home/pgc29/scratch60/helen_mixed_infection/dataraw/GCF_000195955.2_ASM19595v2_genomic.fna \
    -R /home/pgc29/scratch60/helen_mixed_infection/dataraw/targets_NC_000962.3.tsv.gz \
    -a FMT/AD  -Ou |
    bcftools call -A -m --prior 1e-2 -Ou --ploidy 1 |
    bcftools annotate -x INFO,^FORMAT/GT,^FORMAT/AD -Oz  -o {_output}
```

**Comment by BoPeng** [comment]
Houston, winter storm, I think you understand why I did not respond in time.

Anyway, this looks like a bug in submitting tasks to remote host when changes do not trigger a hash change of the task. I will have a look and try to fix it.

**Comment by BoPeng** [comment]
I did some testing and it appears that the bug has been fixed with the patch to #1434.
